01.multithreading/MultiThreading.Task4.Threads.Join/Program.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * 4.	Write a program which recursively creates 10 threads.
+ * Each thread should be with the same body and receive a state with integer number, decrement it,
+ * print and pass as a state into the newly created thread.
+ * Use Thread class for this task and Join for waiting threads.
+ * 
+ * Implement all of the following options:
+ * - a) Use Thread class for this task and Join for waiting threads.
+ * - b) ThreadPool class for this task and Semaphore for waiting threads.
+ */
+
+using System;
+using System.Threading;
+
+namespace MultiThreading.Task4.Threads.Join
+{
+    class Program
+    {
+        public static SemaphoreSlim semaphore = new SemaphoreSlim(1);
+        static void Main(string[] args)
+        {
+            Console.WriteLine("4.	Write a program which recursively creates 10 threads.");
+            Console.WriteLine("Each thread should be with the same body and receive a state with integer number, decrement it, print and pass as a state into the newly created thread.");
+            Console.WriteLine("Implement all of the following options:");
+            Console.WriteLine();
+            
+            Console.WriteLine("- a) Use Thread class for this task and Join for waiting threads.");
+            int number = 10;
```

Comment:
The variable name is not descriptive. Consider using `threadCount`, `iterationAmount` or any other option.

Reply:
done